### PR TITLE
Plugin to redact xml-defined blockly from instructions

### PR DESCRIPTION
**Pull request by bethanyaconnor**
This plugin uses a pretty simple regex to find xml-defined blockly from authored_hints, long_instructions, and short_instructions and replaces it with the block type (suggestions welcome for what to replace the xml with). This PR also adds the plugin to the translation pipeline.

I tried to work off of the tree generated by running the text through the html tokenizer but that ended up just creating sibling nodes for each part of the html, which would've been difficult to iterate through cleanly.

### Example 
Redactable content:
```Connect it underneath the <xml><block type="Dancelab_atTimestamp" block-text="after 4 measures"><title name="TIMESTAMP">4</title><title name="UNIT">"measures"</title></block></xml> event block. Your dancer should start a new move after the fourth measure.```

Redacted:
```Connect it underneath the [Dancelab\_atTimestamp][1] event block. Your dancer should start a new move after the fourth measure.```

Translated:
```Conéctelo debajo del bloque de eventos [Dancelab\_atTimestamp][1]. Tu bailarín debe comenzar un nuevo movimiento después del cuarto compás.```

Restored translation:
```Conéctelo debajo del bloque de eventos <xml><block type="Dancelab_atTimestamp" block-text="after 4 measures"><title name="TIMESTAMP">4</title><title name="UNIT">"measures"</title></block></xml>. Tu bailarín debe comenzar un nuevo movimiento después del cuarto compás.```

### CODE REVIEW

**Comment by Hamms** [comment]
How does this work for instructions like https://studio.code.org/s/coursee-2018/stage/21/puzzle/2 that contain multiple blocks in a single XML tag, or for instructions like https://studio.code.org/s/coursed-2019/stage/18/puzzle/2 that contain multiple XML tags in a single paragraph?

**Comment by bethanyaconnor** [comment]
Instructions with multiple blocks would use the first block. Paragraphs with multiple blocks would create a new redaction for each.

Unfortunately, the first of the examples you linked has a bunch of extra whitespace and newlines in the XML tags so I need to do some work to account for that.

**Comment by Hamms** [comment]
@bethanyaconnor I updated the plugin to (I think) account for all the cases we care about. I also wrote [a simple unit test](https://gist.github.com/Hamms/77d2838f7bec3c020c90885971cef22b), but cannot for the life of me figure out where this should live.

I think I'm going to create a new repo for all our remark plugins; including both the plugins used only by the i18n sync and those used by the markdown renderer.

File: bin/i18n/plugins/blockly.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  // Eat to end of closing tag
+  const original = value.substring(0, closingIndex + closingTag.length);
+  return eat(original)({
+    type: 'redaction',
+    redactionType: BLOCKLY,
+    original,
+    children: [
+      {
+        type: 'text',
+        value: 'blockly block'
```
Comment (bethanyaconnor) [inline bin/i18n/plugins/blockly.js:69]: So all redactions will just say "blockly block"? Is it worth trying to be more descriptive or would that be too confusing?

**Review by bethanyaconnor — COMMENTED** [review]
The biggest case I was struggling with was xml in a separate paragraph, which it looks like this does correctly.

Thanks for bringing this across the finish line!

File: bin/i18n/plugins/blockly.js (hunk 1 above)
Comment (Hamms) [reply bin/i18n/plugins/blockly.js:69]: Yep!

Note that we won't use whatever value translators put in here anywhere, so I debated adding any text at all. I think "blockly block" is sufficiently descriptive, but we could easily change it in the future if we decide we want something more. I was also thinking of just using the string "blockly"

For example, consider https://studio.code.org/s/coursea-2018/stage/4/puzzle/2. The text:

`To get the bird to the pig, snap the <xml><block type="maze_moveEast"/></xml> block to the bottom of the <xml><block type="when_run"/></xml> block, then press "▶ Run"!`

will be redacted to:

`To get the bird to the pig, snap the [blockly block][0] block to the bottom of the [blockly block][1] block, then press "▶ Run"!`

Comment (bethanyaconnor) [reply bin/i18n/plugins/blockly.js:69]: That makes sense. Thanks for the explanation!

**Comment by Hamms** [comment]
> I think I'm going to create a new repo for all our remark plugins; including both the plugins used only by the i18n sync and those used by the markdown renderer.

Added the new repo: https://github.com/code-dot-org/remark-plugins/ and imported this plugin (plus that unit test) here: https://github.com/code-dot-org/remark-plugins/pull/2

**Pull request closed without merging**